### DISCUSSION TITLE
ng add schematic

### DISCUSSION THREAD

**Discussion by p-m-p** [Legal-and-General/canopy#18] [Ideas]
Would be good to have an `ng add @legal-and-general/canopy` schematic setup to install peer dependency of date-fns and to optionally add the library css file to angular.json. This is possibly something i can pick up in the next couple of weeks if you're happy for me to do so. Currently the usage guide does not mention date-fns in the npm install example.

**Comment by andy-polhill** [comment]
Nice, that would be great @p-m-p. I'm not entirely convinced the `date-fns` peer dependency is setup in the best way, I think it loosely follows this, but it does often surprise people when it fails.
https://github.com/ng-packagr/ng-packagr/blob/master/docs/dependencies.md

I haven't worked with schematics yet but one thing I was wondering is if we could get it to override the default app component with a Canopy version of it, at least on a new project. So when the app is spun up the default is a nice Canopy page not the default Angular one.

Also creating components via the CLI, it would be great if it did some of the donkey work, e.g. adds the story file, creates the module etc. 

So many great things we could do, but one step at a time, more than happy if you were able to look at getting it started.

**Comment by p-m-p** [comment]
Yep that sounds ideal. I'd like to be able to extract some configuration out so we can reuse/extend things like tsconfig and tslint in our projects. I think what we might want here is a schematic for adding Canopy to a project as mentioned and a additionally a set of schematics for developing canopy compliant applications.

Something like two additional packages `@legal-and-general/dev-kit` and `@legal-and-general/canopy-schematics` perhaps.

**Comment by andy-polhill** [comment]
@p-m-p just converting to a discussion as I think it's a good idea and worth having around, but I'm trying to clear out the bugs. 🐛
